pkg/cache/cache_test.go

```diff
@@ -0,0 +1,204 @@
+package cache
+
+import "testing"
+
+func Test_Cache_Lifecycle(t *testing.T) {
+	var cac Interface[string, int]
+	{
+		cac = New[string, int]()
+	}
```

Comment:
This lifecycle test covers the basic usage to ensure the interface works sufficiently well.